### PR TITLE
Fix some proto descriptions

### PR DESCRIPTION
**Pull request by craigbox**
A bunch of tidying up:

* put Telemetry API in its place; remove a heading that caused rendering errors
* `s/Pilot/istiod/`
* fix a comment that caused page to wrap
* made some fields be `code` in some descriptions

 Generated files are using the old HTML generator at this point.

### CODE REVIEW

File: networking/v1alpha3/destination_rule.proto [hunk 1]
```diff
@@ -717,7 +719,7 @@ message OutlierDetection {
   int32 consecutive_errors = 1 [deprecated=true];
 
   // Determines whether to distinguish local origin failures from external errors. If set to true
-  // consecutive_local_origin_failure is taken into account for outlier detection calculations.
+  // `consecutive_local_origin_failure` is taken into account for outlier detection calculations.
```
Comment (hzxuzhonghu) [inline networking/v1alpha3/destination_rule.proto:722]: Maybe change to consecutiveLocalOriginFailure, which is used by users

File: networking/v1alpha3/destination_rule.proto [hunk 2]
```diff
@@ -738,11 +740,11 @@ message OutlierDetection {
   // events qualify as a gateway error.
   // This feature is disabled by default or when set to the value 0.
   //
-  // Note that consecutive_gateway_errors and consecutive_5xx_errors can be
+  // Note that `consecutive_gateway_errors` and `consecutive_5xx_errors` can be
```
Comment (hzxuzhonghu) [inline networking/v1alpha3/destination_rule.proto:743]: same for the below fields

File: networking/v1alpha3/destination_rule.proto (hunk 1 above)
Comment (craigbox) [reply networking/v1alpha3/destination_rule.proto]: Thanks, picked up a few more too. 
I would recommend someone goes through each doc and does a proper revision, but I'm just fixing obvious things now.

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]
nice!

**Pull request merged**